webapp/src/components/assetEditor/assetPalette.tsx

```diff
@@ -1,139 +1,228 @@
-        let paletteOptions = AllPalettes.slice();
-
-        if (!paletteOptions.some(p => p.id === currentPalette.id)) {
-            paletteOptions.unshift(currentPalette)
-        }
-
-        const actions: ModalAction[] = [
-            { label: lf("Reset"), onClick: onReset, leftIcon: 'icon undo', className: 'palette-transparent-button', disabled: disableButtons },
-            { label: lf("Save"), onClick: onSave, className: 'green palette-save-button', disabled: disableButtons }
-        ];
-
-        const exitActions: ModalAction[] = [
-            { label: lf("Exit"), onClick: onExit, className: 'teal' }
-        ];
-
-        return <div>
-            <Modal title={lf("Project Color Palette")} onClose={onModalClose} actions={actions}>
-                <PalettePicker
-                    palettes={paletteOptions}
-                    selectedId={currentPalette.id}
-                    onPaletteSelected={onPaletteEdit} />
-                <PaletteEditor palette={currentPalette} onPaletteChanged={onPaletteEdit} />
-            </Modal>
-            {showExitModal && <Modal title={lf("Exit Without Saving")} onClose={onGoBack} actions={exitActions}>
-                <div>{lf("Exit without saving? Your palette changes will be reverted.")}</div>
-            </Modal>}
-        </div>
+    const onNameDone = () => {
+        setShowNameModal(false);
+        setInvalidName(false);
     }
 
-    const getCurrentPalette = () => {
-        if (currentColors) {
-            for (const palette of AllPalettes) {
-                if (isSameAsCurrentColors(palette.colors)) return palette;
-            }
+    const setName = (name: string) => {
+        name = name.trim();
+        if (name.length === 0) {
+            setInvalidName(true);
+            return
+        } else {
+            setInvalidName(false);
         }
-
-        return {
-            id: "custom",
-            name: lf("Custom Palette"),
-            colors: currentColors
-        };
+        setCustomPalettes({
+            ...customPalettes,
+            palettes: {
+                ...customPalettes.palettes,
+                [currentPalette.id]: {
+                    ...currentPalette,
+                    name: name
+                }
+            }
+        });
+        setCurrentPalette({ ...currentPalette, name: name });
     }
 
-    const isSameAsCurrentColors = (colorSet: string[]) => {
+    const isSameColors = (colorSet1: string[], colorSet2: string[]) => {
         let isEqual = true;
-        for (let i = 0; i < colorSet.length; i++) {
-            if (currentColors[i].toLowerCase() !== colorSet[i].toLowerCase()) {
+        for (let i = 0; i < colorSet1.length; i++) {
+            if (colorSet1[i].toLowerCase() !== colorSet2[i].toLowerCase()) {
                 isEqual = false;
                 break;
             }
         }
         return isEqual;
     }
 
-    return renderPaletteModal();
+    const initiatePalettes = () => {
```

Comment:
nit: probably `initializePalettes`? When I see initiate I would think of kicking off some process vs instantiating initial state or something